File: src/main/java/org/itmo/isLab1/batchimporthistory/BatchImportHistoryController.java
package org.itmo.isLab1.batchimporthistory;

import lombok.RequiredArgsConstructor;

import org.itmo.isLab1.batchimporthistory.dto.BatchImportHistoryDto;
import org.itmo.isLab1.users.UserService;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;
import org.springframework.data.web.PageableDefault;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(value = "/api/batch-import/history", produces = MediaType.APPLICATION_JSON_VALUE)
@RequiredArgsConstructor
public class BatchImportHistoryController {
    private final BatchImportHistoryService historyService;
    private final UserService userService;

    @GetMapping
    @PreAuthorize("hasRole('ADMIN') or hasRole('USER')")
    public ResponseEntity<Page<BatchImportHistoryDto>> getHistory(
            @PageableDefault(size = 20, sort = "createdAt", direction = Sort.Direction.DESC) Pageable pageable) {
        boolean isAdmin = userService.getCurrentUser().isAdmin();
        Page<BatchImportHistoryDto> history = historyService.getHistory(pageable, isAdmin);
        return ResponseEntity.ok()
                .header("X-Total-Count", String.valueOf(history.getTotalElements()))
                .body(history);
    }
}

File: frontend/src/controllers/BatchImportHistoryController.ts
import api from './api';

import { AxiosResponse } from 'axios';
import { BatchImportHistoryDto } from '../interfaces/dto/batchimport/BatchImportHistoryDto';
import type Paged from '@/interfaces/models/Paged'

export default class BatchImportHistoryController {
  static async fetchHistory(currentPage, pageSize): Promise<AxiosResponse<Paged<BatchImportHistoryDto>>> {
    return api.get<Paged<BatchImportHistoryDto>>('/batch-import/history', {
      params: {
        page: currentPage - 1,
        size: pageSize,
        sort: 'createdAt,desc'
      }
    });
    
  }
}